### PR TITLE
chore: remove some redudant copies

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

**Review by JakubFornadel — CHANGES_REQUESTED** [review]
Lets wait for my branch to merged, you are touching a lot of stuff that I already refactored

File: src/network/taraxa_capability.cpp [hunk 1]
```diff
@@ -576,7 +576,7 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
         std::vector<Vote> send_next_votes_bundle;
         for (auto const &v : next_votes_bundle) {
           if (!peer->isVoteKnown(v.getHash())) {
-            send_next_votes_bundle.emplace_back(v);
+            send_next_votes_bundle.push_back(std::move(v));
```
Comment (gaoqi8649) [inline src/network/taraxa_capability.cpp:579]: I think emplace_back(v) is some kind same with push_back(std::move(v)). What's different with them?

Comment (MatusKysel) [reply src/network/taraxa_capability.cpp:579]: so emplace should be used when you want to create object in place and insert it to the container. Here we want to move that object so we need to do `push_back(std::move(v))` or `emplace_back(std::move(v))` this should be same, but I prefer to use emplace only when we creating specific object

Comment (gaoqi8649) [reply src/network/taraxa_capability.cpp:579]: Thanks!

File: src/network/tarcap/packets_handlers/vote_packets_handler.cpp [hunk 2]
```diff
@@ -182,7 +182,7 @@ inline void VotePacketsHandler::processPbftNextVotesPacket(const dev::RLP &packe
       std::vector<Vote> send_next_votes_bundle;
       for (auto const &v : next_votes) {
         if (!peer_to_share_to.second->isVoteKnown(v.getHash())) {
-          send_next_votes_bundle.emplace_back(v);
+          send_next_votes_bundle.push_back(std::move(v));
```
Comment (JakubFornadel) [inline src/network/tarcap/packets_handlers/vote_packets_handler.cpp:185]: btw Vote class does not support move semantics. This is true for almost all(if not literally all) of our basic classes like Transaction, DagBlock, etc... So we are in fact doing copies almost everywhere. We should refactor those classes to actually gain something by using std::move

**Review by JakubFornadel — APPROVED** [review]

**Pull request merged**